Trivago/Jade/Application/JsonApi/Request/RequestFactory.php

```diff
@@ -459,11 +456,12 @@ private function createRelationshipBag($data, $dataAsObject, $acceptNull)
             }
             $ids = [];
             foreach ($rawRelationship as $key => $singleRelationship) {
-               $this->validateRelationshipType(
-                   $singleRelationship,
-                   $relationshipName,
-                   sprintf('data.relationships.%s.data[%d].type', $relationshipName, $key)
-               );
+                $this->validateKeys('data.relationships.'.$relationshipName.'.data[]', $singleRelationship, ['id', 'type']);
```

Comment:
Please change data[] to data[$key] so the user knows in which key they have the issue